src/code-considerations/using-unstable-lang/const-evaluation.md

```diff
@@ -0,0 +1,100 @@
+# Const evaluation
+
+Rust's compile-time code evaluation story is still in an experimental state,
+with a large amount of unstable surface area, including areas where there is
+active design work.
+
+The standard library (as of this writing) uses some amount of hacks to work
+around language limitations for some implementations: this document provides a
+framework for evaluating the cost/benefit analysis of using such workarounds to
+enable adding new `const` functions/traits or `const`-ifying existing APIs.
+
+The overall high-level guidance is to evaluate the tradeoff between enabling
+experimentation in users of the standard library (or discovery of limitations
+to these language features) and the costs to the maintenance and readability of
+the standard library. (This is broad guidance applicable to almost any feature,
+but const generics are particularly notable in terms of their potentially wide
+scope).
+
+These are good areas to consider in such an evaluation. You should think about
+whether the given PR is adding value to outweigh its costs along these
+dimensions, and add to the PR description what your justification for the usage
+is. Consider including future plans for the feature as well: if the feature is
+experimental, does it have a clear path toward stabilization in the current
+form?
+
+## Maintainability
+
+Does making this API const require that code be hand-written which was previously automatic?
+
+Does the new code require duplicating, for example through `#[cfg(bootstrap)]`,
+increasing the risk of diverging between the two implementations (and
```

Comment:
afaik this has never happened before. tests are run on `not(bootstrap)` mode, and that is the only relevant mode afaict.

Reply:
I think it's pretty clear that duplicating ~100+ lines of code is a recipe for editing only one of those copies. We don't always have a regression test (e.g., for performance improvements), and due to git these edits may happen entirely in parallel in terms of PRs.